TKeazirian.HTTPServer/Handler/EchoBodyHandler.cs

```diff
@@ -21,9 +27,19 @@ public Response HandleResponse(Request requestObject)
         return responseBuilder.BuildNewResponse(responseStatusLine, responseHeaders, responseBody);
     }
 
-    private string HandleStatusLine()
+    private int HandleStatusCode()
+    {
+        return (int)HttpStatusCode.Ok;
+    }
+
+    private string HandleResponseText(int responseStatusCode)
+    {
+        return GetStatusMessage(responseStatusCode);
+    }
+
+    public string HandleStatusLine(string httpVersion, int responseStatusCode, string responseStatusText)
     {
-        return Constants.Status200;
+        return httpVersion + Constants.Space + responseStatusCode + Constants.Space + responseStatusText;
```

Comment:
there is nothing specific to this Handler here, and this `HandleStatusLine` code could probably be shared across handlers? Could that be moved into the Interface? (it may need to change into an Abstract Class).

Reply:
alternatively, could it be moved into the Response itself? looks like potentially the only thing you would need to know is the `HttpStatusCode` and the rest could be calculated.

Reply:
I like the idea of moving it into the Response itself.  Do you happen to have 30 min to pair with me on this?  I think it's part of the OO programming that I'm having trouble grasping and I want to really understand this.